src/producer.rs

```diff
@@ -1412,6 +1470,29 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_plain_producer() {
```

Comment:
Also test the case where we have a gcno and a gcda and check that the function panics